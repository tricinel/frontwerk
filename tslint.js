module.exports = require('./dist/config/tslintconfig');
