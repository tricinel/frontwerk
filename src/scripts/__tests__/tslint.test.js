/* eslint-disable global-require */
import {
  unquoteSerializer,
  winPathSerializer
} from '../../helpers/serializers';

jest.mock('cross-spawn');
jest.mock('jest');

const cases = require('jest-in-case');
const jestSerializerPath = require('jest-serializer-path');

expect.addSnapshotSerializer(unquoteSerializer);
expect.addSnapshotSerializer(winPathSerializer);
expect.addSnapshotSerializer(jestSerializerPath);

const testCases = [
  {
    name: 'calls tslint CLI with default args',
    fnCalled: 1
  },
  {
    name: 'does not use built-in config with tslint.json file',
    fileExists: filename => filename === 'tslint.json'
  },
  {
    name: 'does not use built-in config with tslint.yaml file',
    fileExists: filename => filename === 'tslint.yaml'
  },
  {
    name: 'does not use built-in config with --config',
    args: ['--config', './custom-config.js']
  }
];

const testFn = ({
  fileExists = () => false,
  hasPkgProp = () => false,
  args = [],
  fnCalled = 1
}) => {
  const { sync: crossSpawnSyncMock } = require('cross-spawn');
  const originalExit = process.exit;
  const originalArgv = process.argv;
  process.exit = jest.fn();

  Object.assign(require('../../utils/fileExists'), { fileExists });
  Object.assign(require('../../utils/pkg'), { hasPkgProp });
  Object.assign(require('../../utils/resolveBin'), {
    resolveBin: (modName, { executable = modName } = {}) => executable
  });

  try {
    // tests
    process.argv = ['node', '../tslint', ...args];
    crossSpawnSyncMock.mockClear();

    require('../tslint');

    expect(crossSpawnSyncMock).toHaveBeenCalledTimes(fnCalled);

    if (fnCalled > 0) {
      const [firstCall] = crossSpawnSyncMock.mock.calls;
      const [script, calledArgs] = firstCall;
      expect([script, ...calledArgs].join(' ')).toMatchSnapshot();
    }
  } catch (error) {
    throw error;
  } finally {
    // afterEach
    process.exit = originalExit;
    process.argv = originalArgv;
    jest.resetModules();
  }
};

cases('tslint', testFn, testCases);
