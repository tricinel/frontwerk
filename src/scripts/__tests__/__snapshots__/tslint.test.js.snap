// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tslint calls tslint CLI with default args 1`] = `tslint --config ./src/config/tslintconfig.js -p tsconfig.json <PROJECT_ROOT>/**/*.ts{,x}`;

exports[`tslint does not use built-in config with --config 1`] = `tslint --config ./custom-config.js -p tsconfig.json <PROJECT_ROOT>/**/*.ts{,x}`;

exports[`tslint does not use built-in config with tslint.json file 1`] = `tslint -p tsconfig.json <PROJECT_ROOT>/**/*.ts{,x}`;

exports[`tslint does not use built-in config with tslint.yaml file 1`] = `tslint -p tsconfig.json <PROJECT_ROOT>/**/*.ts{,x}`;
